app/models/AuthUser.ts

```diff
@@ -9,23 +10,56 @@ export const AuthUserModel = types
     user: types.optional(types.frozen(), {}),
     access_token: types.optional(types.string, ""),
     id_token: types.optional(types.string, ""),
+    isAuthenticated: types.optional(types.boolean, false),
   })
   .actions(withSetPropAction)
-  .views((self) => ({
-    get isLoggedIn() {
-      return !!self.access_token
-    },
-  })) // eslint-disable-line @typescript-eslint/no-unused-vars
   .actions((self) => ({
-    saveUser: (user: AuthUserSnapshotIn) => {
+    setUser: (user: AuthUserSnapshotIn) => {
       self.setProp("user", user)
     },
-    saveAccessToken: (accessToken: string) => {
+    setAccessToken: (accessToken: string) => {
       self.setProp("access_token", accessToken)
     },
-    saveIdToken: (idToken: string) => {
+    setIdToken: (idToken: string) => {
       self.setProp("id_token", idToken)
     },
+    setIsAuthenticated: (isAuthenticated: boolean) => {
+      self.setProp("isAuthenticated", isAuthenticated)
+    },
+  }))
+  .views(() => ({})) // eslint-disable-line @typescript-eslint/no-unused-vars
+  .actions((self) => ({
+    login: async () => {
```

Comment:
Is there a reason to have these actions (login, logout) separated from the actions above (setUser, setAccessToken, etc)?

Reply:
<img width="1161" alt="image" src="https://github.com/aaronannecchiarico/TaskAppMobile/assets/8761662/43ea8c18-c7d5-4baf-8275-97d62546e801">